dev_guide/compute_resources.adoc

```diff
@@ -1,25 +1,129 @@
+ensure that no projects are using more than is appropriate for the cluster size.
 
-When authoring a pod configuration file, you can optionally specify how much CPU and memory (RAM) each container needs in order to better schedule pods in the cluster and ensure satisfactory performance.
+As a developer, you can also
+set link:#dev-compute-resources[requests and limits on compute resources] at the
+pod and container level.
 
-CPU is measured in units called millicores. Each node in a cluster inspects the operating system to determine the amount of CPU cores on the node, then multiplies that value by 1000 to express its total capacity. For example, if a node has 2 cores, the node's CPU capacity would be represented as 2000m. If you wanted to use 1/10 of a single core, it would be represented as 100m.
+The following sections help you understand how to check on your quota and limit
+range settings, what sorts of things they can constrain, and how you can request
+or limit compute resources in your own pods and containers.
 
-Memory is measured in bytes. In addition, it may be used with SI suffices (E, P, T, G, M, K, m) or their power-of-two-equivalents (Ei, Pi, Ti, Gi, Mi, Ki).
+[[dev-quotas]]
+== Quotas
 
+include::admin_guide/quota.adoc[tag=admin_quota_overview]
+
+[NOTE]
+====
+Quotas are set by cluster administrators and are scoped to a given project.
+====
+
+[[dev-viewing-quotas]]
+=== Viewing Quotas
+
+include::admin_guide/quota.adoc[tag=admin_quota_viewing]
+
+Full quota definitions can be viewed by running `oc export` on the object. The
+following show some sample quota definitions:
+
+include::admin_guide/quota.adoc[tag=admin_quota_sample_definitions]
+
+[[dev-managed-by-quota]]
+=== Resources Managed by Quota
+
+include::admin_guide/quota.adoc[tag=admin_quota_resources_managed]
+
+[[dev-quota-scopes]]
+=== Quota Scopes
+
+include::admin_guide/quota.adoc[tag=admin_quota_scopes]
+
+[[dev-quota-enforcement]]
+=== Quota Enforcement
+
+include::admin_guide/quota.adoc[tag=admin_quota_enforcement]
+If project modifications exceed a quota usage limit, the server denies the
+action. An appropriate error message is returned explaining the quota constraint
+violated, and what your currently observed usage stats are in the system.
+
+[[dev-requests-vs-limits]]
+=== Requests vs Limits
+
+include::admin_guide/quota.adoc[tag=admin_quota_requests_vs_limits]
+
+See link:#dev-compute-resources[Compute Resources] for more on setting requests
+and limits in pods and containers.
+
+[[dev-limit-ranges]]
+== Limit Ranges
+
+include::admin_guide/limits.adoc[tag=admin_limits_overview]
+
+[NOTE]
+====
+Limit ranges are set by cluster administrators and are scoped to a given
+project.
 ====
+
+[[dev-viewing-limit-ranges]]
+=== Viewing Limit Ranges
+
+include::admin_guide/limits.adoc[tag=admin_limits_viewing]
+
+Full limit range definitions can be viewed by running `oc export` on the object.
+The following shows an example limit range definition:
```

Comment:
Suggest: "following example shows a  limit range def" to reduce the number of adjectives before the noun "definition".